TestFoundation/TestFileManager.swift

```diff
@@ -225,41 +225,61 @@ class TestFileManager : XCTestCase {
 
     func test_isWritableFile() {
         let fm = FileManager.default
-        let path = NSTemporaryDirectory() + "test_fileAttributes\(NSUUID().uuidString)"
+        let path = NSTemporaryDirectory() + "test_isWritableFile\(NSUUID().uuidString)"
 
-        XCTAssertTrue(fm.createFile(atPath: path, contents: Data(), attributes: nil))
+        XCTAssertTrue(fm.createFile(atPath: path, contents: Data()))
 
         do {
             let attrs = try fm.attributesOfItem(atPath: path)
             let permissions = attrs[FileAttributeKey.posixPermissions] as! UInt16
-            let fileIsReadableFile = (permissions & S_IWUSR == S_IWUSR)
-            let fmIsReadableFile = fm.isReadableFile(atPath: path)
-            XCTAssertTrue(fileIsReadableFile == fmIsReadableFile)
+            let fileIsWritableFile = (permissions & S_IWUSR == S_IWUSR)
+            let fmIsWritableFile = fm.isWritableFile(atPath: path)
+            XCTAssertTrue(fileIsWritableFile == fmIsWritableFile)
         } catch let e {
             XCTFail("\(e)")
         }
     }
 
     func test_isExecutableFile() {
         let fm = FileManager.default
-        let path = NSTemporaryDirectory() + "test_fileAttributes\(NSUUID().uuidString)"
+        let path = NSTemporaryDirectory() + "test_isExecutableFile\(NSUUID().uuidString)"
 
-        XCTAssertTrue(fm.createFile(atPath: path, contents: Data(), attributes: nil))
+        XCTAssertTrue(fm.createFile(atPath: path, contents: Data()))
 
         do {
             let attrs = try fm.attributesOfItem(atPath: path)
             let permissions = attrs[FileAttributeKey.posixPermissions] as! UInt16
-            let fileIsReadableFile = (permissions & S_IXUSR == S_IXUSR)
-            let fmIsReadableFile = fm.isReadableFile(atPath: path)
-            XCTAssertTrue(fileIsReadableFile == fmIsReadableFile)
+            let fileIsExecutableFile = (permissions & S_IXUSR == S_IXUSR)
+            let fmIsExecutableFile = fm.isExecutableFile(atPath: path)
+            XCTAssertTrue(fileIsExecutableFile == fmIsExecutableFile)
         } catch let e {
```

Comment:
I would modify this test to set the executable bit on the file and then simply test it with:
`XCTAssertTrue(fm.isExecutableFile(atPath: path))` (both before and after setting the bit)

Testing equality this way could mask an issue if the `let fileIsExecutableFile = (permissions & S_IXUSR == S_IXUSR)` was wrong.

Reply:
Makes sense, will do